### PR TITLE
Keep deleted projects name

### PR DESCRIPTION
**Pull request by hsato03**
### Description

When deleting a project, its name is always set to `null`. To maintain the integrity of the project's database history, this behavior was removed.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

I created and deleted a project. Then, I created a SELECT SQL query for the `projects` table and the deleted project's name was there.

### CODE REVIEW

**Comment by kiranchavala** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@hsato03 
if you want this to be in 4.18.1.0, can you rebase with 4.18 ?

**Review by DaanHoogland — COMMENTED** [review]
code looks good, but this behaviour has deliberately been added. Did you investigate why @hsato03 ? it may have undesired side effects. Functionally I agree with you and we may have to deal with any side effects to get this in. There are no DB constraints so let's test, and get it in? cc @shwstppr @weizhouapache

**Review by kiranchavala — APPROVED** [review]
LGTM,

Tested manually 

1. Create a project and delete it 

Before fix after deletion of a project NULL name was updated 

```
mysql> select * from projects \G;
*************************** 1. row ***************************
                id: 1
              name: NULL
              uuid: 336fd0ae-fa12-4e09-82d3-345451001b7c
      display_text: testproject
project_account_id: 7
         domain_id: 1
           created: 2023-08-14 06:29:46
           removed: 2023-08-14 06:30:41
             state: Disabled
1 row in set (0.00 sec)
```



After fix , the name of the project is retained 

```
MariaDB [cloud]> select * from projects \G;
*************************** 1. row ***************************
                id: 1
              name: test
              uuid: 22679e43-ac2a-4094-8d18-dae4f396c2d0
      display_text: test
project_account_id: 5
         domain_id: 1
           created: 2023-08-14 07:01:51
           removed: 2023-08-14 07:05:57
             state: Disabled
```

**Comment by weizhouapache** [comment]
it is a weird behaviour to set project name to null when project is removed.

@DaanHoogland 's concern also makes sense.
I checked the ProjectDaoImpl. it seems this PR will not cause any issue.

**Comment by weizhouapache** [comment]
added to 4.19.0.0 milestone as this PR is targeted to main branch

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Review by rohityadavcloud — APPROVED** [review]
LGTM - need to check/test, if this allows creation of a project of a name which was earlier removed as the DB might has a uniq constraint.

**Comment by DaanHoogland** [comment]
> LGTM - need to check/test, if this allows creation of a project of a name which was earlier removed as the DB might has a uniq constraint.

The DB has no constraint but problems may arise when projects are seached by name somewhere in the code.

**Comment by hsato03** [comment]
> @hsato03 if you want this to be in 4.18.1.0, can you rebase with 4.18 ?

@weizhouapache Of course.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by DaanHoogland** [comment]
not sure if the tests done by @kiranchavala cover all basis, but there is functional agreement that this is the proper functionality, and fallout needs to be dealt with. Shall we merge @weizhouapache ?

**Comment by hsato03** [comment]
> not sure if the tests done by @kiranchavala cover all basis, but there is functional agreement that this is the proper functionality, and fallout needs to be dealt with. Shall we merge @weizhouapache ?

@DaanHoogland, @weizhouapache 
Hi guys, any updates on this one?

**Comment by weizhouapache** [comment]
> > not sure if the tests done by @kiranchavala cover all basis, but there is functional agreement that this is the proper functionality, and fallout needs to be dealt with. Shall we merge @weizhouapache ?
> 
> @DaanHoogland, @weizhouapache Hi guys, any updates on this one?

@hsato03 
code looks ok.
However, we do not look into the side-effect yet. Since the original issue is trivial, I will NOT merge this into 4.18.1.0.
cc @DaanHoogland

**Comment by DaanHoogland** [comment]
@kiranchavala did/can you test the corner cases mentioned? Most notably:
- recreating a project with the same name
- search functionality
thanks

**Comment by kiranchavala** [comment]
@DaanHoogland 

I have tested the corner cases  and its working fine

1. recreating a project with the same name
2. search functionality

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 vmware-67u3

**Review by DaanHoogland — APPROVED** [review]

**Pull request merged**
